### DISCUSSION TITLE
Jaccard distance on a set

### DISCUSSION THREAD

**Discussion by prabh-singh123** [moj-analytical-services/splink#658] [Q&A]
Is it possible to have a custom comparator for a feature that is represented as a set?

I have hashed out the first name column. The hashed column: "first_name_hashes" contains a set of integers. 
![image](https://user-images.githubusercontent.com/105444112/179864897-5c77d651-3393-4540-a47f-87f880568fc0.png)

Can I use Jaccard distance for this column in splink?

**Comment by ericmanning** [comment]
Hi -- not a developer, but a user of this package.

Take a look at `array_intersect_level()` [here](https://github.com/moj-analytical-services/splink/blob/master/splink/spark/spark_comparison_level_library.py). You could represent comparison levels as the number of common elements between two sets.

**Reply by prabh-singh123** [reply]
Hey @ericmanning ! Thanks for your response. I guess I could work with that as an alternative to Jaccard Distance. But, I can't seem to find `array_intersect_level()` for DuckDB backend [here](https://github.com/moj-analytical-services/splink/blob/1b0e52bb20676d76353160a59d46f5a8b9317f92/splink/duckdb/duckdb_comparison_level_library.py). Any leads there?

**Reply by ericmanning** [reply]
Unfortunately there's no `array_intersect` function in duckDB (yet), which is the SQL function underneath `array_intersect_level()`, so you're stuck writing out something on your own that uses their [existing functions](https://duckdb.org/docs/sql/functions/overview).
